torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -778,6 +807,16 @@ c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupNCCL::WorkNCCL::
   return c10::make_intrusive<FutureNCCL>(shared_from_this(), outputs_);
 }
 
+void ProcessGroupNCCL::enqueue(
```

Comment:
(can be add in a followup PR if necessary) I wonder if this enqueue/workCleanupLoop/runLoop can be abstract out and deduped  from ProcessGroupGloo/NCCL and put it into the base class?

Reply:
There is some NCCL-specific logic added to `workCleanupLoop` on in the stack, but we have some ideas on potentially consolidating the looping threads to reduce the overhead of additional threads if we have multiple process groups. I believe the priority for this work item depends on the results from the multiple process groups experiments for TSC.